### PR TITLE
[CP] Fix not being able to hide iOS status bar via setEnabledSystemUIMode

### PR DESCRIPTION
**Pull request by LinXunFeng**
Original PR: https://github.com/flutter/engine/pull/48271

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm [hunk 1]
```diff
@@ -41,15 +41,15 @@
 using namespace flutter;
 
 static void SetStatusBarHiddenForSharedApplication(BOOL hidden) {
-#if APPLICATION_EXTENSION_API_ONLY
+#if not APPLICATION_EXTENSION_API_ONLY
```
Comment (stuartmorgan) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm:44]: I've never seen `not` in a preprocessor directive. Is this a non-standard extension?

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformPluginTest.mm [hunk 2]
```diff
@@ -368,4 +368,93 @@ - (void)testViewControllerBasedStatusBarHiddenUpdate {
...
 
+- (void)testStatusBarHiddenUpdate {
+  id bundleMock = OCMPartialMock([NSBundle mainBundle]);
+  OCMStub([bundleMock objectForInfoDictionaryKey:@"UIViewControllerBasedStatusBarAppearance"])
+      .andReturn(@NO);
+  id mockApplication = OCMClassMock([UIApplication class]);
+  OCMStub([mockApplication sharedApplication]).andReturn(mockApplication);
+
+  // Enabling system UI overlays to update status bar.
+  FlutterEngine* engine = [[FlutterEngine alloc] initWithName:@"test" project:nil];
```
Comment (stuartmorgan) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformPluginTest.mm:379]: This file isn't ARC on `stable`, so all the `alloc` calls need `autorelease`.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm (hunk 1 above)
Comment (LinXunFeng) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm:44]: The macro `not` as alternative representation of `!` in cpp.

Related links are as follows
- [Alternative operator spellings - macro](https://cplusplus.com/reference/ciso646/)
- [‘#if’ directive](https://gcc.gnu.org/onlinedocs/cpp/If.html)

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformPluginTest.mm (hunk 2 above)
Comment (LinXunFeng) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformPluginTest.mm]: done

**Pull request merged**